src/c-opentimelineio/tests/OTIOMediaReferenceTests.cpp

```diff
@@ -0,0 +1,159 @@
+    Any*        decoded = /** allocate memory for destinantion */
+        create_safely_typed_any_serializable_object((SerializableObject*) mr);
+    bool decoded_successfully =
+        deserialize_json_from_string(encoded, decoded, errorStatus);
+    ASSERT_TRUE(decoded_successfully);
+
+    SerializableObject* decoded_object = safely_cast_retainer_any(decoded);
+    EXPECT_TRUE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) mr, decoded_object));
+
+    Any_destroy(mr_any);
+    mr_any = NULL;
+    SerializableObject_possibly_delete(decoded_object);
+    decoded_object = NULL;
+    OTIOErrorStatus_destroy(errorStatus);
+    errorStatus = NULL;
+}
+
+TEST_F(OTIOMediaReferenceTests, FilepathTest)
+{
+    ExternalReference* filepath =
+        ExternalReference_create("var/tmp/foo.mov", NULL, NULL);
+    OTIOErrorStatus* errorStatus = OTIOErrorStatus_create();
+
+    Any* filepath_any = create_safely_typed_any_serializable_object(
+        (SerializableObject*) filepath);
+    const char* encoded =
+        serialize_json_to_string(filepath_any, errorStatus, 4);
+    Any* decoded = /** allocate memory for destinantion */
+        create_safely_typed_any_serializable_object(
+            (SerializableObject*) filepath);
+    bool decoded_successfully =
+        deserialize_json_from_string(encoded, decoded, errorStatus);
+    ASSERT_TRUE(decoded_successfully);
+
+    SerializableObject* decoded_object = safely_cast_retainer_any(decoded);
+    EXPECT_TRUE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) filepath, decoded_object));
+
+    Any_destroy(filepath_any);
+    filepath_any = NULL;
+    SerializableObject_possibly_delete(decoded_object);
+    decoded_object = NULL;
+    OTIOErrorStatus_destroy(errorStatus);
+    errorStatus = NULL;
+}
+
+TEST_F(OTIOMediaReferenceTests, EqualityTest)
+{
+    ExternalReference* filepath =
+        ExternalReference_create("var/tmp/foo.mov", NULL, NULL);
+    ExternalReference* filepath2 =
+        ExternalReference_create("var/tmp/foo.mov", NULL, NULL);
+
+    EXPECT_TRUE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) filepath, (SerializableObject*) filepath2));
+
+    MissingReference* bl = MissingReference_create(NULL, NULL, NULL);
+
+    EXPECT_FALSE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) filepath, (SerializableObject*) bl));
+
+    filepath  = ExternalReference_create("var/tmp/foo.mov", NULL, NULL);
+    filepath2 = ExternalReference_create("var/tmp/foo2.mov", NULL, NULL);
+
+    EXPECT_FALSE(SerializableObject_is_equivalent_to(
+        (SerializableObject*) filepath, (SerializableObject*) filepath2));
+}
+
+TEST_F(OTIOMediaReferenceTests, IsMissingTest)
+{
+    ExternalReference* filepath =
+        ExternalReference_create("var/tmp/foo.mov", NULL, NULL);
+    EXPECT_FALSE(
+        MediaReference_is_missing_reference((MediaReference*) filepath));
```

Comment:
@jminor @meshula @ssteinbach @reinecke 

This is an example of the inheritance stuff I was talking about. Instead of defining a method `ExternalReference_is_missing_reference` I've simply cast the pointer.

After giving it a thought I think it will be convenient for the end user if I wrap the inherited methods too. But I'd like to hear your thoughts.

Reply:
Yes, wrapping inherited methods, in an unambiguous way, is the way to go.